frame/contracts/src/lib.rs

```diff
@@ -725,15 +734,24 @@ pub mod pallet {
 			)
 		}
 
-		/// TODO: add benchmark for base weight
-		/// TODO: add a minimum weight limit so that people don't spam small free extrinsics
 		#[pallet::call_index(9)]
-		#[pallet::weight(*weight_limit)]
+		#[pallet::weight(T::WeightInfo::migrate_noop().saturating_add(T::WeightInfo::migration()).saturating_add(*weight_limit))]
```

Comment:
linking the conversation that is masked 
https://github.com/paritytech/substrate/pull/14045/files/e49f0ce0f6d4d72f4de98afffc1bdb6e61463946#r1184779495